### PR TITLE
Improve thunk call disassembly in R2RDump

### PR DESCRIPTION
**Pull request by trylek**
This change improves the disassembly of thunk-based calls to R2R
helpers used by Crossgen1 and considered for Crossgen2 to improve
disassembly quality and facilitate easier comparison of Crossgen1
vs. Crossgen2 codegen.

I have also added new options for hiding the instruction offsets
in the disassembly and for omitting functions with identical
disassembly in the common functions diff output. For SPC this
reduced the size of the dump in <code>--naked --hide-offsets
--diff-hide-same-disasm</code> mode by about one half.

In view of the fact that Andrew wasn't too happy about my change
to stop logging to the console I further simplified the related
code by basically reverting that aspect of my change in a more
radical manner.

I have also moved the call to CoreDisTools.ClearOutputBuffer
into CoreDisTools.GetInstruction so that we don't need to call it
manually in the higher-level code.

Thanks

Tomas

### CODE REVIEW

File: src/coreclr/src/tools/r2rdump/TextDumper.cs [hunk 1]
```diff
@@ -189,8 +189,8 @@ internal override void DumpRuntimeFunction(RuntimeFunction rtf)
         /// </summary>
         internal override void DumpDisasm(RuntimeFunction rtf, int imageOffset)
         {
-            int indent = (_options.Naked ? 11 : 32);
-            string indentString = new string(' ', indent);
+            int indent = (_options.Naked ? _options.HideOffsets ? 8 : 11 : 32);
+            string indentString = new string('\t', indent / 8) + new string(' ', indent % 8);
```
Comment (janvorli) [inline src/coreclr/src/tools/r2rdump/TextDumper.cs]: I would prefer using spaces. My editing tools have tab set to 4 spaces, some people prefer 8 spaces ...

**Review by janvorli — COMMENTED** [review]
I have just one comment, otherwise it looks good. I am really looking forward for using it.

File: src/coreclr/src/tools/r2rdump/TextDumper.cs (hunk 1 above)
Comment (trylek) [reply src/coreclr/src/tools/r2rdump/TextDumper.cs]: Agreed, fixed in 2nd commit.

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Pull request merged**